### PR TITLE
[Qt6Base] Update to 6.3.0

### PR DESCRIPTION
**Pull request by barche**
First step in updating Qt6Base, building for the host only (see comment at the top of the build_tarballs.jl).

### CODE REVIEW

File: Q/Qt6Base/build_tarballs.jl [hunk 1]
```diff
@@ -66,19 +66,26 @@ case "$target" in
...
 
 # These are the platforms we will build for by default, unless further
 # platforms are passed in on the command line
-platforms = expand_cxxstring_abis(filter(!Sys.isapple, supported_platforms()))
-platforms_macos = [ Platform("x86_64", "macos") ]
+if host_build
+    platforms = [Platform("x86_64", "linux",cxxstring_abi=:cxx11,libc="musl")]
+else
+    platforms = expand_cxxstring_abis(filter(!Sys.isapple, supported_platforms()))
+    platforms_macos = [ Platform("x86_64", "macos") ]
```
Comment (giordano) [inline Q/Qt6Base/build_tarballs.jl:87]: Should also add aarch64?

File: Q/Qt6Base/build_tarballs.jl [hunk 2]
```diff
@@ -134,10 +140,16 @@ dependencies = [
...
 
+if !host_build
+    push!(dependencies, HostBuildDependency("Qt6Base_jll"))
+end
+
 include("../../fancy_toys.jl")
 
-if any(should_build_platform.(triplet.(platforms_macos)))
-    build_tarballs(ARGS, name, version, sources, script, platforms_macos, products_macos, dependencies; preferred_gcc_version = v"8", julia_compat="1.6")
+@static if !host_build
```
Comment (giordano) [inline Q/Qt6Base/build_tarballs.jl:149]: For the record, a top-level `@static if` does nothing more than a simple `if`

File: Q/Qt6Base/build_tarballs.jl (hunk 1 above)
Comment (barche) [reply Q/Qt6Base/build_tarballs.jl:87]: Yes, will do when adding all platforms.

File: Q/Qt6Base/build_tarballs.jl (hunk 2 above)
Comment (barche) [reply Q/Qt6Base/build_tarballs.jl:149]: I got `platforms_macos` undefined with a regular if.

**Pull request merged**